### PR TITLE
Remove dependency on Apache commons lang

### PR DESCRIPTION
**Pull request by elharo**
@ejona86 Apache Commons Lang is a large dependency we barely use, yet it leaks into every project that uses GCP client libraries. It's a fairly easy replacement. Ping me offline for associated support ticket if you're interested.

### CODE REVIEW

**Comment by ejona86** [comment]
I recently noticed this dep in #7733 and also thought it would be good to remove it.

@elharo, the code doesn't compile.

**Comment by elharo** [comment]
Fixed the missing import. Building this project jas so many prerequisites it's easier to push it up to Github and let the CI do the work.

**Comment by ejona86** [comment]
> Building this project jas so many prerequisites

If you take a look at our COMPILING.md, we tell you that all the hard stuff is optional. You can do `./gradlew -PskipCodegen=true -PskipAndroid=true` at which point you only need Java installed beforehand.

File: alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java
Comment (ejona86) [inline alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java:67]: Don't make this a static variable; move it into isRunningOnGcp. It is only used in isRunningOnGcp and isRunningOnGcp is only called once.

File: alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java
Comment (ejona86) [inline alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java:74]: How did you come up with this code? Why not the more conventional `startsWith()` and why not use "windows" here?

Comment (elharo) [reply alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java]: image

I picked "win" instead of "windows" in case we end up with something weird like "Win95" or "WinXP" one of these days. indexOf seemed a little more general than startsWith.

Comment (ejona86) [reply alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java]: But it seems you copied from a lower-ranked answer that initially had bugs like "checking for 'darwin' after 'win'" if you look at the very first comment. And you can't simply remove the Darwin case and expect proper behavior.

OpenJDK uses [very regular names for Windows](https://github.com/openjdk/jdk/blob/91244cc738e92163d99b8951c5d95b546447f341/src/java.base/windows/native/libjava/java_props_md.c#L478). Mac is hard-coded to "Mac OS X" and Unix uses `uname -s`. I don't see the hard-coding for OS X in Java 7, so it probably used "Darwin." That's all obviously OpenJDK-specific, but I'd expect other JDKs to be similar (use uname for most cases, start Windows names with "Windows").

Use `startsWith()` and spell out "windows" in full.

Comment (elharo) [reply alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java]: done

**Review by ejona86 — APPROVED** [review]
Will wait for @jiangtaoli2016's review before merging.

**Review by jiangtaoli2016 — APPROVED** [review]

**Pull request merged**
